gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -193,15 +226,7 @@ private void handleLiveCommit(
                   addContent = !isExpired;
                 }
                 if (addContent) {
-                  Content content = ((Operation.Put) operation).getContent();
-                  bloomFilterMap
-                      .computeIfAbsent(
-                          content.getId(),
-                          k ->
-                              new ContentBloomFilter(
-                                  gcStateParamsPerTask.getBloomFilterSize(),
-                                  gcParams.getBloomFilterFpp()))
-                      .put(content);
+                  bloomFilter.put(((Operation.Put) operation).getContent());
                 }
               });
     }
```

Comment:
oh.. Are you referring to the scenario of where `perReference` cutoff time is configured. And that reference is dropped after sometime.  Also, the `deadReference` cutoff time is not configured.


In that case, Currently GC is using `perReference` cutoff time instead of default cutoff time.
Ok, we can make deadReference cutofftime as mandatory config as you suggested.